crates/torii/core/src/engine.rs

```diff
@@ -169,6 +169,17 @@ pub enum FetchDataResult {
     None,
 }
 
+impl FetchDataResult {
+    pub fn block_id(&self) -> BlockId {
+        match self {
+            FetchDataResult::Range(range) => BlockId::Number(range.latest_block_number),
+            FetchDataResult::Pending(_pending) => BlockId::Tag(BlockTag::Pending),
+            // we dont require block_id when result is none, so this is a dummy value
+            FetchDataResult::None => BlockId::Number(0),
```

Comment:
won't it be more easy to maintain having an option or something? Just t be sure it's not use in a non intended manner?

But not blocking for this one, could be updated after.

Reply:
good point, will update.